Improve docs of leaderelection configuration (#20601)
elastic/beats: metricbeat/docs/running-on-kubernetes.asciidoc

[float]
==== Kubernetes deploy manifests

You deploy {beatname_uc} as a https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/[DaemonSet]
to ensure that there's a running instance on each node of the cluster. These
instances are used to retrieve most metrics from the host, such as system
metrics, Docker stats, and metrics from all the services running on top of
Kubernetes.

In addition one Pod of the Daemonset will hold a leader lock each time which makes it responsible for
handling cluster wide monitoring.
This instance is used to retrieve metrics that are unique for the whole
cluster, such as Kubernetes events or
https://github.com/kubernetes/kube-state-metrics[kube-state-metrics].

> You canfind more information about leader election configuration options at <<configuration-autodiscover>>.

Reviewer: ```suggestion
You can find more information about leader election configuration options at <<configuration-autodiscover>>.
```

[float]
==== Kubernetes deploy manifests

You deploy {beatname_uc} as a https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/[DaemonSet]
to ensure that there's a running instance on each node of the cluster. These
instances are used to retrieve most metrics from the host, such as system
metrics, Docker stats, and metrics from all the services running on top of
Kubernetes.

> In addition one Pod of the Daemonset will hold a leader lock each time which makes it responsible for

Reviewer 2: ```suggestion
In addition, one of the Pods in the DaemonSet will constantly hold a _leader lock_ which makes it responsible for
```

> handling cluster wide monitoring.

Reviewer 2: ```suggestion
handling cluster-wide monitoring.
```

This instance is used to retrieve metrics that are unique for the whole
cluster, such as Kubernetes events or
https://github.com/kubernetes/kube-state-metrics[kube-state-metrics].
You can find more information about leader election configuration options at <<configuration-autodiscover>>.

> Note: If you are upgrading from older versions of these manifests please make sure that there are no redundant parts

Reviewer 2: ```suggestion
Note: If you are upgrading from older versions, please make sure there are no redundant parts
```

> as left-overs from previous versions. Deployment specification and its ConfigMaps might be the case.

Reviewer 2: ```suggestion
as left-overs from the old manifests. Deployment specification and its ConfigMaps might be the case.
```

[float]
==== Deploying Metricbeat to collect cluster level metrics for big clusters

> In some cases, users might want to monitor really bog Kubernetes clusters. In such cases,

Reviewer: ```suggestion
In some cases, users might want to monitor really big Kubernetes clusters. In such cases,
```

the Pod that will be collecting cluster level metrics might face performance issues due to
resources limitations. In this case users might consider avoid using leader election configuration
for the Daemonset and instead deploy a singleton Metribceat instance using a
https://github.com/elastic/beats/blob/7.9/deploy/kubernetes/metricbeat/metricbeat-deployment.yaml[Deployment] with the

> proper https://github.com/elastic/beats/blob/7.9/deploy/kubernetes/metricbeat/metricbeat-deployment-configmap.yaml[configuration].

Reviewer: This is linking to older branches, we may easily forget to apply changes there that could be needed for future versions. Would it make sense to somehow copy the important parts of these manifests to some doc file? Or we just remove these links and leave to the user the definition of their configuration?

Author: 🤔 My intention here is to point to the latest version that had been supporting the Deployment approach just for reference. I don't see any reason of updating this however do you see any case I might miss?

Reviewer: If this is going to be the documentation for big clusters I think we should keep it updated. Imagine in 7.10/7.11, someone with a big cluster sees this section, and they use the 7.9 deployment linked here. They may end up using an incorrect version, and they may also be missing some configuration change that we haven't backported.

Author: 👍  gotcha, thanks!

[float]

> ==== Deploying Metricbeat to collect cluster level metrics for big clusters

Reviewer 2: ```suggestion
==== Deploying Metricbeat to collect cluster-level metrics in large clusters
```

> In some cases, users might want to monitor really big Kubernetes clusters. In such cases,

Reviewer 2: ```suggestion
The size and the number of nodes in a Kubernetes cluster can be fairly large at times, and in such cases
```

the Pod that will be collecting cluster level metrics might face performance issues due to

> resources limitations. In this case users might consider avoid using leader election configuration

Reviewer 2: ```suggestion
resources limitations. In this case users might consider to avoid using the leader election strategy
```

> for the Daemonset and instead deploy a singleton Metribceat instance using a Deployment with the

Reviewer 2: ```suggestion
and instead run a dedicated, standalone Metribceat instance using a Deployment in addition to the DaemonSet.
```

> proper configuration.

Reviewer 2: ```suggestion
```
